hal/inc/sleep_hal.h

```diff
@@ -83,18 +89,18 @@ typedef enum hal_sleep_wait_t {
 
 #if PLATFORM_ID > PLATFORM_GCC
 static_assert(sizeof(hal_sleep_mode_t) == 1, "length of hal_sleep_mode_t should be 1-bytes aligned.");
-static_assert(sizeof(hal_wakeup_source_type_t) == 2, "length of hal_wakeup_source_type_t should be 2-bytes aligned.");
+static_assert(sizeof(hal_wakeup_source_type_t) == 4, "length of hal_wakeup_source_type_t should be 4-bytes aligned.");
 static_assert(sizeof(hal_sleep_wait_t) == 1, "length of hal_sleep_wait_t should be 1-bytes aligned.");
 #endif
 
 /**
  * HAL sleep wakeup source base
  */
+typedef struct hal_wakeup_source_base_t hal_wakeup_source_base_t;
```

Comment:
Why double typedefs?

Reply:
C compiler will throw an error that the `hal_wakeup_source_base_t*` in this struct is an incomplete type.

Reply:
You can avoid this by using adding `struct` when defining `next` as well FYI.